### ISSUE TITLE
Can't include CSS files

### ISSUE THREAD

**Issue by DanHulton** [mongeslani/kohana-less#9]
I've found one of the real useful bits of this module is including separate files and having it compile them all down into one CSS file.  However, I have a lot of existing CSS files being updated by third parties, and renaming them isn't an attractive option.

Instead, it would be nice if raw CSS files were allowed to be compiled as well.  The nice thing about LESS is that it EXTENDS CSS, so raw CSS files should compile just as well as LESS files.

To that end, I modified a few functions:

```
<?php defined('SYSPATH') or die('No direct script access.');

/**
 * Modification of Less_Core to allow for CSS files to be "compiled" as well.
 */
class Less extends Less_Core {
    /**
     * Get the link tag of less paths
     *
     * @param   array     array of css paths
     * @param   string    value of media css type
     * @param   boolean   allow compression
     * @return  string    link tag pointing to the css paths
     */
    public static function compile($array = array(), $media = 'screen')
    {
        // return comment if array is empty
        if (empty($array)) return self::_html_comment('no less files');

        $stylesheets = array();
        $assets = array();

        // validate
        foreach ($array as $file)
        {
            // remove extension if its present
            $file = preg_replace('/\.less/', '', $file);
            if (file_exists($file.'.less'))
            {
                array_push($stylesheets, $file . '.less');
            }
            else
            {
                // Might be a CSS file
                $file = preg_replace('/\.css/', '', $file);
                if (file_exists($file . '.css')) {
                    array_push($stylesheets, $file . '.css');
                }
                // Welp, couldn't find it
                else {
                    Kohana::$log->add("ERROR", "Could not find LESS/CSS file: $file");
                    array_push($assets, self::_html_comment('could not find '.Kohana::debug_path($file).'.less'));
                }
            }
        }

        // all stylesheets are invalid
        if ( ! count($stylesheets)) return self::_html_comment('all less files are invalid');

        // get less config
        $config = Kohana::config('less');

        // if compression is allowed
        if ($config['compress'])
        {
            return html::style(self::_combine($stylesheets), array('media' => $media));
        }

        // if no compression
        foreach ($stylesheets as $file)
        {
            $filename = self::_get_filename($file, $config['path']);
            array_push($assets, html::style($filename, array('media' => $media)));
        }

        return implode("\n", $assets);
    }

    /**
     * Check if the asset exists already, if not generate an asset
     *
     * @param   string   path of the css file
     * @return  string   path to the asset file
     */
    protected static function _get_filename($file, $path)
    {
        // get the filename
        $filename = preg_replace('/^.+\//', '', $file);

        // get the last modified date
        $last_modified = self::_get_last_modified(array($file));

        // compose the expected filename to store in /media/css
        $compiled = $filename.'-'.$last_modified.'.css';

        // compose the expected file path
        $filename = $path.$compiled;

        // if the file exists no need to generate
        if ( ! file_exists($filename))
        {
            // create data holder
            $data = '';

            touch($filename, filemtime($file) - 3600);

            lessc::ccompile($file, $filename);
        }

        return $filename;
    }

    /**
     * Combine the files
     *
     * @param   array    array of asset files
     * @return  string   path to the asset file
     */
    protected static function _combine($files)
    {
        // get assets' css config
        $config = Kohana::config('less');

        // get the most recent modified time of any of the files
        $last_modified = self::_get_last_modified($files);

        // compose the asset filename
        $compiled = md5(implode('|', $files)).'-'.$last_modified.'.css';

        // compose the path to the asset file
        $filename = $config['path'].$compiled;

        // if the file exists no need to generate
        if ( ! file_exists($filename))
        {
            self::_generate_assets($filename, $files);
        }

        return $filename;
    }

    /**
     * Generate an asset file
     *
     * @param   string   filename of the asset file
     * @param   array    array of source files
     */
    protected static function _generate_assets($filename, $files)
    {
        // create data holder
        $data = '';

        touch($filename);

        ob_start();

        foreach($files as $file)
        {
            $data .= file_get_contents($file);
        }

        echo $data;

        file_put_contents($filename, ob_get_clean(), LOCK_EX);

        self::_compile($filename);
    }

    /**
     * Get the most recent modified date of files
     *
     * @param   array    array of asset files
     * @return  string   path to the asset file
     */
    protected static function _get_last_modified($files)
    {
        $last_modified = 0;

        foreach ($files as $file) 
        {
            $modified = filemtime($file);
            if ($modified !== false and $modified > $last_modified) $last_modified = $modified;
        }

        return $last_modified;
    }
}
```

In short, in the `compile()` function, I check for less files, and if they exist, add them to the `$stylesheets` array with the ".less" extension appended **here** instead of every other time throughout the file.  Then I do the same thing with CSS files.

The other functions I included because a) I removed the ".less" extension append, or b) they were part of Less_Core and called `self::somefunction()`, which called the un-fixed variant in Less_Core, not the fixed one in Less.
